lib/grac/client.rb

```diff
@@ -25,13 +25,24 @@ def initialize(uri, options = {})
         :postprocessing => options[:postprocessing] || {},
         :middleware     => options[:middleware]     || []
       }
-      @options.freeze
+
+      if options[:postprocessing]
+        options[:postprocessing]
+          .each_with_object(patterns = { postprocessing: {} }) do |(key, value), obj|
+          if !key.kind_of?(Regexp)
+            obj[:postprocessing][Regexp.new(key)] = value
+          end
+        end
+        options.merge!(patterns)
```

Comment:
Looks good, although I am wondering:
* `patterns` will only contain `Regexp` + `transformation lambda`: Wouldn't `patterns = {}` and `obj[Regexp.new(key)] = value` be sufficient?
* Could it be more efficient to have an else condition instead of a merge later on:
```ruby
if !key.kind_of?(Regexp)
  obj[:postprocessing][Regexp.new(key)] = value
else
  obj[:postprocessing][key] = value
end
```
* If you'd skip the merge part (only values with `Regexp` keys should make it into patterns then) you could probably skip the `delete_if` part
* Does it make sense to initialize `@options` with `:postprocessing => options[:postprocessing] || {},`? if `:postprocessing` is not null, it will be overwritten, otherwise it will be `{}`, so maybe `:postprocessing => {},` would be more fitting now to cover the default case
* Did you remove `@options.freeze` on purpose or by accident?

Reply:
Suggestion:
```ruby
@options = {
  :connecttimeout => options[:connecttimeout] || 0.1,
  :timeout        => options[:timeout]        || 15,
  :params         => options[:params]         || {},
  :headers        => {
    "User-Agent"   => "Grac v#{Grac::VERSION}",
    "Content-Type" => "application/json;charset=utf-8"
  }.merge(options[:headers] || {}),
  :postprocessing => {},
  :middleware     => options[:middleware]     || []
}

if options[:postprocessing]

  options[:postprocessing]
    .each_with_object(postprocessing = {}) do |(key, value), obj|
    if key.kind_of?(Regexp)
      obj[key] = value
    else
      obj[Regexp.new(key)] = value
    end
  end
  
  @options[:postprocessing] = postprocessing
end
@options.freeze
```

Reply:
I agree with your suggestions and i just made the corresponding changes. To address some of your points:

> patterns will only contain Regexp + transformation lambda: Wouldn't patterns = {} and obj[Regexp.new(key)] = value be sufficient?

Now that we are going to do `@options[:postprocessing] = postprocessing` i think it makes more sense to initialize an empty hash.

> Did you remove @options.freeze on purpose or by accident?

Yes 😊. Fixed now.

I also renamed some variables: `key -> pattern`, `value -> transformation` and inside `#postprocessing` i removed `patterns` and just take and iterate on `@options[:postprocessing].keys` directly.